backend/billing/views.py

```
@@ -0,0 +1,709 @@
+        if billing_cycle not in ['monthly', 'annual']:
+            return JsonResponse({'success': False, 'error': 'Invalid billing cycle'}, status=400)
+        
+        # Get base price
+        amount = PLAN_PRICING[plan_type][billing_cycle]
+        
+        # Apply discount if provided
+        discount_percentage = 0
+        if discount_code:
+            # Check for referral codes (50% off first month for monthly plans)
+            if discount_code.startswith('REF_') and billing_cycle == 'monthly':
+                discount_percentage = 50
```

Ruff (0.14.3)

Local variable `discount_percentage` is assigned to but never used

Remove assignment to unused variable `discount_percentage`

(F841)